src/jit/morph.cpp

```diff
@@ -10855,10 +10850,7 @@ GenTreePtr          Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* ma
                 }
                 else if (andOp->gtType == TYP_LONG)
                 {
-                    if (andOp->gtOp.gtOp2->gtOper != GT_CNS_NATIVELONG)
-                        goto SKIP;
-
-                    if (andOp->gtOp.gtOp2->gtIntConCommon.LngValue() != 1)
+                    if (!andOp->IsIOne())
```

Comment:
Here too we are checking whether andOp is cns one instead what we should be checking is second operand of andOp